lib/internal/validators.js

```diff
@@ -258,6 +258,21 @@ function validateUnion(value, name, union) {
   }
 }
 
+function validateLinkHeaderValue(value, name) {
+  const linkValueRegExp = /^((<([^>]*)>;)\s*(rel="?([^;]*)"?;?)\s*((as|anchor|title)="?(.*)"?)?)$/;
```

Comment:
```suggestion
  const linkValueRegExp = /^(?:<[^>]*>;)\s*(?:rel="?[^;]*"?;?)\s*(?:(?:as|anchor|title)="?.*"?)?$/;
```

There seems to be no need to use capturing groups. Also this matches stuff like this `</styles.css>; rel="` and I'm not sure if it should. I think that is invalid.

Finally, according to [recheck](https://makenowjust-labs.github.io/recheck/), the regular expression is vulnerable to ReDoS but we are in a sever context so I think it does not matter much.

Reply:
thanks @lpincawe can solve the optional quotes by using a backreference, I updated the regex pattern and updated a test suite for the regex tests -> https://regexr.com/6rv4u